utils_linux.go

```diff
@@ -121,13 +121,10 @@ func setupIO(process *libcontainer.Process, container *libcontainer.Container, c
 			if err != nil {
 				return nil, err
 			}
-			uc, ok := conn.(*net.UnixConn)
-			if !ok {
-				return nil, errors.New("casting to UnixConn failed")
-			}
-			t.postStart = append(t.postStart, uc)
-			socket, err := uc.File()
+			t.postStart = append(t.postStart, conn)
+			socket, err := conn.(*net.UnixConn).File()
 			if err != nil {
+				conn.Close()
```

Comment:
FWIW I prefer the `Err error` pattern for stuff like this, but for this one case it's fine...

Reply:
There's only one error path here that has/needs `conn.Close`. Anyway, fixed.